src/compressed_tensors/quantization/lifecycle/forward.py

```diff
@@ -60,9 +58,31 @@ def fake_quantize(
     bit_range = 2**args.num_bits
     max_q = torch.tensor(bit_range / 2 - 1, device=x.device)
     min_q = torch.tensor(-bit_range / 2, device=x.device)
-    Q = torch.zeros_like(x)
-    Q = quantize(x, scale, zero_point, min_q, max_q)
-    return dequantize(Q, scale, zero_point)
+
+    columns = x.shape[1]
+    group_size = args.group_size
+
+    if group_size is None or group_size == 0:
+        Q = quantize(x, scale, zero_point, min_q, max_q)
+        DQ = dequantize(Q, scale, zero_point)
+
+    elif group_size > 0:
+        DQ = torch.zeros_like(x)
+
+        # TODO: vectorize the for loop
+        # TODO: fix genetric assumption about the tensor size for computing group
+        for i in range(int(columns / group_size)):
```

Comment:
do we have a check that group size divides the columns evenly?

Reply:
No its expected that those matches. 
If there is a validation, it should happen in upstream anyways, no there

For i can do ceil( column / group_size)